purchase_deposit/models/purchase.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2019 Elico Corp, Dominique K. <dominique.k@elico-corp.com.sg>
+# Copyright 2019 Ecosoft Co., Ltd., Kitti U. <kittiu@ecosoft.co.th>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    def copy_data(self, default=None):
+        if default is None:
+            default = {}
+        default["order_line"] = [
+            (0, 0, line.copy_data()[0])
+            for line in self.order_line.filtered(lambda l: not l.is_deposit)
```

Comment:
why run two for when you can do it in one for?

```suggestion
            for line in self.order_line if not line.is_deposit
```